### PR TITLE
refactor: Feature flags getter

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
The Feature flags getter has should be returned an explicit boolean value. The unit test makes a minor refactor.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts [hunk 1]
```diff
@@ -16,48 +16,41 @@
...
-    1,
-    'Failed to query feature flag ALLOW_DASHBOARD_DOMAIN_SHARDING (see error below)',
+  mockConsole();
+  Object.defineProperty(window, 'featureFlags', {
+    value: undefined,
+  });
+
+  expect(
+    isFeatureEnabled(FeatureFlag.ALLOW_DASHBOARD_DOMAIN_SHARDING),
+  ).toBeFalsy();
```
Comment (villebro) [inline superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts:30]: If we're expecting a `boolean` value (which this PR is making more specific), I'd prefer to make the assertion more specific, too:
```suggestion
  expect(
    isFeatureEnabled(FeatureFlag.ALLOW_DASHBOARD_DOMAIN_SHARDING),
  ).toEqual(false);
```

File: superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts [hunk 2]
```diff
@@ -16,48 +16,41 @@
...
-    false,
-  );
+  Object.defineProperty(window, 'featureFlags', {
+    value: {},
+  });
 
-  reset();
+  expect(
+    isFeatureEnabled(FeatureFlag.ALLOW_DASHBOARD_DOMAIN_SHARDING),
+  ).toBeFalsy();
```
Comment (villebro) [inline superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts:45]: same here

File: superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts [hunk 3]
```diff
@@ -16,48 +16,41 @@
...
-  };
+  Object.defineProperty(window, 'featureFlags', {
+    value: {
+      CLIENT_CACHE: true,
+    },
+  });
 
-  expect(isFeatureEnabled(FeatureFlag.CLIENT_CACHE)).toEqual(true);
-  reset();
+  expect(isFeatureEnabled(FeatureFlag.CLIENT_CACHE)).toBeTruthy();
```
Comment (villebro) [inline superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts:55]: and here

**Review by villebro — COMMENTED** [review]
Some code style opinions, LMKWYT

File: superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts (hunk 1 above)
Comment (zhaoyongjie) [reply superset-frontend/packages/superset-ui-core/test/utils/featureFlag.test.ts:30]: Good point. I mistakenly thought that `toBeFalsy()` is a shortcut for `toEqual(false)`. will update

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**